### ISSUE TITLE
More precise code ranges for JET underlines

### ISSUE THREAD

**Issue by jariji** [julia-vscode/julia-vscode#3443]
Given this code

```jl
f(x::Integer) = x
g(x::AbstractFloat) = let
    z = zz
    y = f(x)
    z + y
end
```

`JET.report_file(path; analyze_from_definitions=true)`  produces

![image](https://github.com/julia-vscode/julia-vscode/assets/96840304/e1ccd86a-dc81-4248-9fef-68d31e90bfb9)

The whole line 3 and 4 are underlined but the problem is really the `zz` and `f(x)` subexpressions, not the whole line. It would be nice to have more precise range underlines to help isolate the problem.

JET v0.8.12
Julia VSCode 1.54.2

**Comment by pfitzseb** [comment]
Yes, this is a JET issue. Our API supports column info, but JET currently can't emit that (due to `Expr` not having them).
